backend/sirius-web-spring-collaborative/src/main/java/org/eclipse/sirius/web/spring/collaborative/projects/EditingContextEventProcessor.java

```diff
@@ -183,42 +188,41 @@ private void publishEvent(IInput input, Optional<IPayload> optionalPayload) {
             EventHandlerResponse response = optionalResponse.get();
 
             this.disposeRepresentationIfNeeded();
+            this.refreshRepresentations(representationId, response.getChangeKind());
 
-            // @formatter:off
-            this.representationEventProcessors.values().stream()
-                .filter(representationEventProcessor -> {
-                    /**
-                     * Filter the representationEventProcessor related to the IRepresentationInput in order to
-                     * make sure that only other representations are refreshed by the response.
-                     */
-                    if (input instanceof IRepresentationInput) {
-                        IRepresentationInput representationInput = (IRepresentationInput) input;
-                        UUID currentRepresentationId = representationInput.getRepresentationId();
-                        UUID aRepresentationId = representationEventProcessor.getRepresentation().getId();
-                        return !Objects.equals(currentRepresentationId, aRepresentationId);
-                    } else {
-                        return true;
-                    }
-                })
-                .filter(representationEventProcessor -> {
-                    IRepresentation representation = representationEventProcessor.getRepresentation();
-                    return response.getShouldRefreshPredicate().test(representation);
-                })
-                .forEach(representationEventProcessor -> {
-                    representationEventProcessor.refresh();
-                    IRepresentation representation = representationEventProcessor.getRepresentation();
-                    this.applicationEventPublisher.publishEvent(new RepresentationRefreshedEvent(this.editingContext.getId(), representation));
-                });
-            // @formatter:on
-
-            if (response.isEditingContextDirty()) {
+            if (this.shouldPersistTheEditingContext(response.getChangeKind())) {
                 this.editingContextPersistenceService.persist(this.editingContext);
             }
         }
 
         return optionalResponse;
     }
 
+    /**
+     * Refresh all the representations except the one with the given representationId.
+     *
+     * @param representationId
+     *            The identifier of the representation which should not be refreshed
+     * @param changeKind
+     *            The kind of change to consider in order to determine if the representation should be refreshed
+     */
+    private void refreshRepresentations(UUID representationId, ChangeKind changeKind) {
```

Comment:
I know the semantics of the arguments is documented, but the signature is misleading/easy to misuse. `refreshOtherRepresentations` maybe?

Reply:
I did not like the name either... I'll change that